File: project04/n_factorial.rb
# n_factorial computes the factorial of n
#
# Haim Hong
# Date Feb 4 2022
#
def factorial (n)
    answer = 1.0
    for a in 2..n do # loop until it reaches n
        answer *= a # multiply answer by a that increments by 1 each loop
    end
    return answer.to_s
end
if __FILE__ == $0
    print "Enter the number to computer the factorial of it: " # ouputs the string
    n = gets.to_f # gets input for n for n!
    puts "\nThe logarithms of #{n.to_s} is #{factorial(n)}" # outputs the result of it

end

File: project04/NFactorial.java
/* 
 * Outputs the factorial of number that use has inputed
 * 
 * Completed by: Haim Hong
 * Feb 3 2022
 ********************************************************/
import java.util.Scanner; 		// Get input

class NFactorial{

  public static void main(String[] args){
    Scanner keyboard = new Scanner(System.in); // scanners for input
    System.out.println("Enter the number to compute the factorial of it: "); // outputs string

    double n;
    n = keyboard.nextDouble(); // gets input n factorial
    System.out.println(factorial(n)); // outputs the factorial of n
}
public static double factorial(double n){ // function to compute factorial of n
    double answer = 1.0;
    for (double i = 2; i <= n;i++){ // loop until reaches the n while incrementing by 1
        answer *= i; // multiply answer by i 
    }
        return answer; // returns the result
    }
}
